fastddsspy_participants/src/cpp/visualization/ModelParser.cpp

```diff
@@ -211,68 +211,282 @@ ComplexParticipantData ModelParser::participants(const SpyModel& model, const dd
 }
 
 
+void set_complex_endpoint(
+    ComplexEndpointData& result,
+    const eprosima::spy::participants::EndpointInfo& endpoint)
+{
+    result.guid = endpoint.guid;
+    result.topic.topic_name = endpoint.topic.m_topic_name;
+    result.topic.topic_type = endpoint.topic.type_name;
+    if (endpoint.topic.topic_qos.durability_qos)
+    {
+        result.qos.durability = "transient-local";
+    }
+    else
+    {
+        result.qos.durability = "volatile";
+    }
+    if (endpoint.topic.topic_qos.reliability_qos)
+    {
+        result.qos.reliability = "best-effort";
+    }
+    else
+    {
+        result.qos.reliability = "reliable";
+    }
+}
+
 std::vector<SimpleEndpointData> ModelParser::writers(const SpyModel& model)
 {
     std::vector<SimpleEndpointData> result;
-    // TODO
+
+    for (const auto& endpoint : model.endpoint_database_)
+    {
+        if (endpoint.second.is_writer())
+        {
+            std::string participant_name = "";
+            // Look for participant name
+            for (const auto& participant : model.participant_database_)
+            {
+                if (participant.first.guid_prefix() == endpoint.second.guid.guid_prefix())
+                {
+                    participant_name = participant.second.name;
+                    break;
+                }
+            }
+            result.push_back({
+                endpoint.second.guid,
+                participant_name,
+                {
+                    endpoint.second.topic.m_topic_name,
+                    endpoint.second.topic.type_name
+                }
+            });
+        }
+    }
+
     return result;
 }
 
 std::vector<ComplexEndpointData> ModelParser::writers_verbose(const SpyModel& model)
 {
     std::vector<ComplexEndpointData> result;
-    // TODO
+
+    for (const auto& endpoint : model.endpoint_database_)
+    {
+        if (endpoint.second.is_writer())
+        {
+            result.push_back(writers(model, endpoint.second.guid));
+        }
+    }
+
     return result;
 }
 
 ComplexEndpointData ModelParser::writers(const SpyModel& model, const ddspipe::core::types::Guid& guid)
 {
     ComplexEndpointData result;
-    // TODO
+
+    for (const auto& endpoint : model.endpoint_database_)
+    {
+        if (endpoint.first == guid && endpoint.second.is_writer())
+        {
+            // Look for participant name
+            for (const auto& participant : model.participant_database_)
+            {
+                if (participant.first.guid_prefix() == guid.guid_prefix())
+                {
+                    result.participant_name = participant.second.name;
+                    break;
+                }
+            }
+            set_complex_endpoint(result, endpoint.second);
+
+            break;
+        }
+    }
+
```

Comment:
This is also repeated, let's move it to generic functions that are common to readers and writers.
I would use a function that receives in one argument the EntityType, and from it choose only those endpoints with such type (a not use is_reader() or is_writer())